[Liu Jihan] iP

Liu-Jihan opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

LiuJihan committed:
Level 0. Rename, Greet, Exit

LiuJihan committed:
Level 2. Add, List

LiuJihan committed:
Revert "Level 2. Add, List"

This reverts commit 97561a3b041237536706ff493b9778010dc2d885.

LiuJihan committed:
add while loop to echo the input , and exit by "bye" (Level 1. Echo)

LiuJihan committed:
add an arrayList to memorize (Level 2. Add, List)

LiuJihan committed:
Class Task has been created

LiuJihan committed:
logic been changed and used Task instead of string(Level-3)

LiuJihan committed:
logic has been changed, subclass of task such as ddl,event, and todo has been added into the function

LiuJihan committed:
Update the automated test case(A-TextUiTesting)

LiuJihan committed:
ctry to catch exceptions(Level-5)

LiuJihan committed:
add delete(level-6)

LiuJihan committed:
Add storage.java
Task and subclass : Modify to be able to recongnise the file from storage
OscarL: Use storage drom duke.txt

LiuJihan committed:
Merge branch 'branch-Level-7'

Windofbitter commented on src/main/java/OscarL.java:
```diff
@@ -80,6 +81,7 @@ public static void main(String[] args) {
                         tasks.add(newDeadline);
```
I like the way you organize the code, it is really clean with the trailing comment. May be we can try to separate the add task logic unit with the ui output logic by a few line to make it more organized?

Windofbitter commented on src/main/java/OscarL.java:
```diff
@@ -94,6 +96,22 @@ public static void main(String[] args) {
+                        break;
+
+                    case "delete": // Remove a task
+                        if (inputParts.length < 2) {
```
It is good that you follow coding standard strictly by adding four space for statements like conditions, it is a common mistake.

Windofbitter commented on src/main/java/Task.java:
```diff
@@ -21,4 +24,46 @@ public void markAsNotDone() {
+        boolean isDone = parts[1].equals("1");
+        String description = parts[2];
+
+        switch (type) {
```
Maybe we can try to align with the coding standard that switch and case should have the same indentation?

Windofbitter commented on src/main/java/Task.java:
```diff
@@ -21,4 +24,46 @@ public void markAsNotDone() {
+                    throw new IllegalArgumentException("Invalid deadline format in file.");
+                }
+                Deadline deadline = new Deadline(description, parts[3]);
+                if (isDone) deadline.markAsDone();
```
Maybe we can also wrap the single statement with brackets as well to follow the coding standard?

LiuJihan committed:
scratch is used for testing function, not used for any purposes.

LiuJihan committed:
OscarL: Import several time package
Event & Deadline : Use partern format to recognise time and used time package instead string for date

LiuJihan committed:
Merge branch 'branch-Level-8'

LiuJihan committed:
Add Ui to store welcome message
Add Command class for all the store execution
OscarL:Logic has been simplified

LiuJihan committed:
Merge branch 'branch-A-MoreOOP'

LiuJihan committed:
Use package to organise this

LiuJihan committed:
Merge branch 'add-gradle-support'

Casielim commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,110 @@
+        String command = parts[0];
+
+        switch (command) {
+            case "bye":
```
IIRC, there is no need to indent the case clauses according to Java coding standards. You might want to adjust that.

Casielim commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,110 @@
+        return isExit;
+    }
+
+    public static Command parse(String input, TaskList tasks, Ui ui, Storage storage) throws OscarLException {
```
Try to avoid long methods, as they often contain more information, making debugging harder later. Consider shortening them if possible.

Casielim reviewed (commented):
Looks good to me.
Here are some suggestions that won't affect the functionality of your code.

LiuJihan committed:
Add gradle to automate test Deadline and TaskList.

LiuJihan committed:
Create a JAR File for OscarL

LiuJihan committed:
Add JavaDoc to all classes and methods

LiuJihan committed:
Merge branch 'branch-A-JavaDoc'

LiuJihan committed:
All classes: Modify to follow java coding standard

LiuJihan committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/task/Task.java
#	src/main/java/task/TaskList.java

LiuJihan committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/task/Task.java
#	src/main/java/task/TaskList.java

LiuJihan committed:
Add find book(Level-9)

LiuJihan committed:
Merge branch 'branch-Level-9'

LiuJihan committed:
Solve conflicts

varuuuun commented on src/main/java/oscarl/OscarL.java:
```diff
@@ -0,0 +1,63 @@
+ */
+public class OscarL {
+    private Storage storage;
+    private TaskList tasks;
```
The `TaskList` object you are referrencing to here is a single object, not the list itself - hence it should be named as `task` or `taskList` instead of `tasks`?

varuuuun commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,118 @@
+
+        switch (type) {
+            case TODO_TYPE:
+                ToDo todo = new ToDo(description);
```
Inconsistent naming? if class name is `ToDo`, then shouldn't object name be `toDo`?

varuuuun commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,133 @@
+     * @return The size of the task list.
+     */
+
+    public int size() {
```
Should be named as `getSize()`. Use verb instead of noun.

varuuuun commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,174 @@
+     * @return The parsed Command object.
+     * @throws OscarLException If the input command is invalid.
+     */
+    public static Command parse(String input, TaskList tasks, Ui ui, Storage storage) throws OscarLException {
```
Method name might be a bit generic, you might want to consider `parseStringToCommand` instead?

varuuuun reviewed (commented):
Some nits on naming.

Liu-Jihan committed:
Update README.md

Liu-Jihan committed:
Update README.md

Liu-Jihan committed:
Update README.md

LiuJihan committed:
GFMD is updated

LiuJihan committed:
Merge branch 'master' of https://github.com/Liu-Jihan/ip

* 'master' of https://github.com/Liu-Jihan/ip:
  Update README.md
  Update README.md
  Update README.md

LiuJihan committed:
implement JavaFX

LiuJihan committed:
Merge branch 'branch-Level-10'

LiuJihan committed:
Main & Command : assertion has been implemented

LiuJihan committed:
Refactor: Break Down Long parse() Method in Command Class

The `parse()` method in the `Command` class was excessively long, making it difficult to maintain and debug. This commit refactors the method by extracting multiple helper methods, improving code structure and readability.

### Changes Implemented:
- **Extracted individual methods** for each command (`mark`, `unmark`, `delete`, `todo`, `deadline`, `event`, `list`, `bye`).
- **Simplified `parse()` method** into a clean `switch-case` that delegates to helper methods.
- **Added assertions** to validate input, ensuring robustness and preventing invalid operations.
- **Maintained existing functionality** without altering program behavior.

### Benefits:
✔️ **Easier maintenance** – Modifying a single command logic is now more straightforward.
✔️ **Improved readability** – Code is now well-structured and modular.
✔️ **Enhanced debugging** – Errors are easier to trace and fix.
✔️ **Encapsulation** – Each method now handles one specific command, following clean code principles.

All existing tests pass successfully, and functionality remains unchanged.

Liu-Jihan committed:
Merge pull request #2 from Liu-Jihan/branch-A-Assertions

Main & Command : assertion has been implemented

LiuJihan committed:
solve some random quality issue in MainWindow

Liu-Jihan committed:
Revert "Main & Command : assertion has been implemented"

Liu-Jihan committed:
Merge pull request #3 from Liu-Jihan/revert-2-branch-A-Assertions

Revert "Main & Command : assertion has been implemented"

LiuJihan committed:
Merge branch 'master' of https://github.com/Liu-Jihan/ip

* 'master' of https://github.com/Liu-Jihan/ip:
  Revert "Main & Command : assertion has been implemented"
  Main & Command : assertion has been implemented

Liu-Jihan committed:
Merge pull request #4 from Liu-Jihan/branch-A-CodeQuality

 Branch a code quality

LiuJihan committed:
Merge branch 'master' of https://github.com/Liu-Jihan/ip

* 'master' of https://github.com/Liu-Jihan/ip:
  solve some random quality issue in MainWindow
  Refactor: Break Down Long parse() Method in Command Class

LiuJihan committed:
places class is added, use addplace & removeplace&  listplace to checkout

LiuJihan committed:
Merge tag 'BCD-Extension'

LiuJihan committed:
update Readme

Liu-Jihan committed:
Update README.md

Liu-Jihan committed:
Update README.md

LiuJihan committed:
Update build

LiuJihan committed:
Update deadline method

LiuJihan committed:
Update event

LiuJihan committed:
Update event

LiuJihan committed:
Add javadoc

LiuJihan committed:
Add javadoc

LiuJihan committed:
add javadoc

LiuJihan committed:
add javadoc

LiuJihan committed:
Update GFMD
